documentation/cluster-controller/cluster-controller.adoc

```diff
@@ -0,0 +1,296 @@
+= Cluster Controller
+
+The cluster controller is in charge of deploying a Kafka cluster alongside a Zookeeper ensemble. It's also able to deploy a
+Kafka Connect cluster which connects to an existing Kafka cluster. Such a cluster can be deployed using the Source2Image OpenShift
+feature as well in order to include more connectors plugins.
+
+When the cluster controller is up and running, it starts to "watch" for ConfigMaps containing the Kafka or Kafka Connect
+cluster configuration. Such ConfigMaps need to have a specific label which is, by default, `strimzi.io/kind=cluster`
+(as described <<Format of the cluster ConfigMap,later>>) that can be changed through a corresponding environment variable.
+
+When a new ConfigMap is created in the Kubernetes/OpenShift cluster, the controller gets the cluster configuration from
+its `data` section and starts creating a new Kafka or Kafka Connect cluster by creating the necessary Kubernetes/OpenShift
+resources, such as StatefulSets, ConfigMaps, Services etc.
+
+Every time the ConfigMap is updated by the user with some changes in the `data` section, the controller performs corresponding
+updates on the Kubernetes/OpenShift resources which make up the Kafka or Kafka Connect cluster. Resources are either patched
+or deleted and then re-created in order to make the Kafka or Kafka Connect cluster reflect the state of the cluster ConfigMap.
+This might cause a rolling update which might lead to service disruption.
```

Comment:
> which might lead to service disruption.

I think we should elaborate on this somewhere. I guess the thing about rolling updates of Kafka pods currently is we don't wait for brokers to catch up (i.e. join the ISR) before restarting the next one. We should explain this, I think. Likewise with ZK, we should explain the kind of disruption the user should expect.